### ISSUE TITLE
Pretraining unstable

### ISSUE THREAD

**Issue by liuhl-source** [dandelin/ViLT#7]
I carefully processed the pretraining datasets. But the loss is unstable when pretraining. Which is not monotone decreasing. 
Is this right?  I worry that there are some mistakes in the complicated data preprocessing.
For example:

Epoch 0:  34%|███▍      | 12412/36673 [4:26:06<8:40:09,  1.29s/it, loss=2.58, v_num=0]

Epoch 0:  35%|███▍      | 12665/36673 [4:36:41<8:44:29,  1.31s/it, loss=3.12, v_num=0]

Epoch 0:  39%|███▉      | 14368/36673 [5:43:41<8:53:33,  1.44s/it, loss=2.57, v_num=0]

Epoch 0:  40%|███▉      | 14583/36673 [5:56:04<8:59:23,  1.47s/it, loss=3.36, v_num=0]

**Comment by dandelin** [comment]
HI, the loss can be increased under 25k training steps since the learning rate is warmed up until then.

**Comment by liuhl-source** [comment]
> HI, the loss can be increased under 25k training steps since the learning rate is warmed up until then.

Thanks for your reply!
